### PR TITLE
Update rb help documentation

### PR DESCRIPTION
**Pull request by kyleknap**
Doc update specifically targets behavior with versioned objects.

cc @jamesls @mtdowling @rayluo

### CODE REVIEW

File: awscli/customizations/s3/subcommands.py [hunk 1]
```diff
@@ -593,7 +598,10 @@ class MbCommand(S3TransferCommand):
 
 class RbCommand(S3TransferCommand):
     NAME = 'rb'
-    DESCRIPTION = "Deletes an S3 bucket."
+    DESCRIPTION = (
+        "Deletes an empty S3 bucket. A bucket must be completely empty "
+        "of objects and versioned objects before it can be deleted."
```
Comment (jamesls) [inline awscli/customizations/s3/subcommands.py]: This makes is seem like you can't use `rb` if you have non-versioned objects in your bucket.  Should this be reworded?  Or maybe `--force` should be mentioned here.

Comment (kyleknap) [inline awscli/customizations/s3/subcommands.py]: Yeah I will mention ``--force``.

**Comment by kyleknap** [comment]
@jamesls I added the note about ``--force`` in the ``rb`` help page. Should be good to be looked at again.

**Comment by jamesls** [comment]
:shipit:

**Pull request merged**